posthog/models/kafka_debug/sql.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass
+from posthog.clickhouse.kafka_engine import kafka_engine
+from posthog.clickhouse.table_engines import MergeTreeEngine
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+
+
+@dataclass
+class KafkaDebugKafkaTable:
+    brokers: list[str]
+    topic: str
+    consumer_group: str = "debug"
+
+    @property
+    def table_name(self) -> str:
+        return f"kafka_{self.topic}_debug"
+
+    def get_create_table_sql(self) -> str:
+        return f"""
+      CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+      (
+        payload String
+      )
+      ENGINE={kafka_engine(kafka_host=",".join(self.brokers), topic=self.topic, group=self.consumer_group)}
+      SETTINGS input_format_values_interpret_expressions=0, kafka_skip_broken_messages = 100
+    """
+
+    def get_drop_table_sql(self) -> str:
+        return f"""
+      DROP TABLE IF EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+    """
+
+
+@dataclass
+class KafkaDebugTable:
+    topic: str
+
+    @property
+    def table_name(self) -> str:
+        return f"{self.topic}_debug"
+
+    def get_create_table_sql(self) -> str:
+        engine = MergeTreeEngine(self.table_name, ver="timestamp")
+        return f"""
+      CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}' (
+        payload String,
+        _timestamp DateTime,
+        _timestamp_ms Nullable(DateTime64(3)),
+        _partition UInt64,
+        _offset UInt64
```

Comment:
That shouldn't be required here since we are just grabbing the payload right? It's not actually going to try to deserialize the JSON

Reply:
I'm not sure to be honest, since we are defining the Kafka engine with a JSONEachRow. I don't know if ClickHouse will try to validate that every row is a valid JSON before ingesting it.

Reply:
Oh ok! 💯  In that case then no problem, I just was not completely sure.